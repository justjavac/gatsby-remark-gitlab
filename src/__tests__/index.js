const Remark = require(`remark`);
const plugin = require(`../index`);

const remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
})

test(`mention`, () => {
  const markdownAST = remark.parse(`@user`);
  plugin({ markdownAST });
  expect(markdownAST).toMatchSnapshot();
});

test(`issues`, () => {
  const markdownAST = remark.parse(`#12`);
  plugin({ markdownAST });
  expect(markdownAST).toMatchSnapshot();
});

test(`hash`, () => {
  const markdownAST = remark.parse(`a5c3785ed8d6a35868bc169f07e40e889087fd2e`);
  plugin({ markdownAST });
  expect(markdownAST).toMatchSnapshot();
});
