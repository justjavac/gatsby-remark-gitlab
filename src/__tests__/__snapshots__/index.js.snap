// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hash 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "inlineCode",
              "value": "a5c3785",
            },
          ],
          "title": null,
          "type": "link",
          "url": "https://gitlab.com/justjavac/gatsby-remark-gitlab/commit/a5c3785ed8d6a35868bc169f07e40e889087fd2e",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 41,
          "line": 1,
          "offset": 40,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 41,
      "line": 1,
      "offset": 40,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`issues 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
            "indent": Array [],
            "start": Object {
              "column": 2,
              "line": 1,
              "offset": 1,
            },
          },
          "type": "text",
          "value": "12",
        },
      ],
      "depth": 1,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 1,
          "offset": 3,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "heading",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 1,
      "offset": 3,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`mention 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "@user",
                },
              ],
              "type": "strong",
            },
          ],
          "title": null,
          "type": "link",
          "url": "https://gitlab.com/user",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 6,
          "line": 1,
          "offset": 5,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 6,
      "line": 1,
      "offset": 5,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
